app/views/course/forum/topics/_controls.html.slim

```diff
@@ -0,0 +1,39 @@
+span.btn-group
+  - if @topic.subscribed_by?(current_user)
+    = link_to unsubscribe_course_forum_topic_path(current_course, @forum, @topic),
+              title: t('course.forum.topics.unsubscribe.tag'), class: ['btn', 'btn-danger'],
+              method: :delete do
+      = fa_icon 'heart-o'.freeze
+  - else
+    = link_to subscribe_course_forum_topic_path(current_course, @forum, @topic),
+              title: t('course.forum.topics.subscribe.tag'), class: ['btn', 'btn-success'],
+              method: :post do
+      = fa_icon 'heart'.freeze
+
+span.pull-right.btn-group
+  - if can?(:edit, @topic)
+    - if @topic.locked
```

Comment:
I just learned from V1. In V1 the one who can edit can also set lock. So do we need a new permission like `set_lock` in V2? If so, I'll do that.